R/PipeOpFDASmooth.R

```diff
@@ -0,0 +1,91 @@
+#' @title Smoothing Functional Data
+#' @name mlr_pipeops_fda.smooth
+#'
+#' @description
+#' Smoothes functional data using [`tf::tf_smooth()`].
+#'
+#' @section Parameters:
+#' The parameters are the parameters inherited from [`PipeOpTaskPreprocSimple`], as well as the following
+#' parameters:
+#' * `affect_columns` :: `function` | [`Selector`][mlr3pipelines::Selector] \cr
+#'   [`Selector`][mlr3pipelines::Selector] function, takes a `Task` as argument and returns a `character`
+#'   of features to keep. The flattening is only applied to those columns.\cr
+#'   See [`Selector`][mlr3pipelines::Selector] for example functions. Default is
+#'   selector_all()`, which selects all of the `functional` features.
+#' * `method` :: `character(1)`\cr
+#'   One of "lowess", "rollmean", "rollmedian", "savgol".
+#'   The default is "lowess". All all other methods ignore non-equidistant arg values.
+#' * `args` :: named `list()`\cr
+#'   List of named arguments that is passed to `tf_smooth()`. See the help page of `tf_smooth()` for
+#'   default values.
+#' * `verbose` :: `logical(1)`\cr
+#'   Whether to print messages during the transformation.
+#'   Is initialized to `FALSE`.
+#'
+#' @export
+#' @examples
+#' library(mlr3pipelines)
+#' task = tsk("fuel")
+#' po_smooth = po("fda.smooth", method = "rollmean", args = list(k = 5))
+#' task_smooth = po_smooth$train(list(task))[[1L]]
+#' task_smooth
+#' task_smooth$data(cols = c("NIR", "UVVIS"))
+PipeOpFDASmooth = R6Class("PipeOpFDASmooth",
+  inherit = mlr3pipelines::PipeOpTaskPreprocSimple,
+  public = list(
+    #' @description Initializes a new instance of this Class.
+    #' @param id (`character(1)`)\cr
+    #'   Identifier of resulting object, default `"fda.smooth"`.
+    #' @param param_vals (named `list`)\cr
+    #'   List of hyperparameter settings, overwriting the hyperparameter settings that would
+    #'   otherwise be set during construction. Default `list()`.
+    initialize = function(id = "fda.smooth", param_vals = list()) {
+      check_method = crate(function(x) {
+        msg = check_choice(x, c("lowess", "rollmean", "rollmedian", "savgol"))
+        if (!isTRUE(msg)) return(msg)
+
+        switch(x,
+          savgol = require_namespaces("pracma", quietly = TRUE),
+          rollmedian = require_namespaces("zoo", quietly = TRUE),
+          rollmean = require_namespaces("zoo", quietly = TRUE),
```

Comment:
Is that required? The zoo and pracma are imports of the `tf ` package: https://github.com/tidyfun/tf/blob/dev/DESCRIPTION#L28